data/src/main/java/com/smile/data/repository/CharacterRepositoryImpl.kt

```diff
@@ -12,19 +13,20 @@ class CharacterRepositoryImpl @Inject constructor(
     private val localDataSource: CharacterLocalDataSource
 ) : CharacterRepository {
 
-    override suspend fun getCharacterList(): Wrapper<List<Character>> {
+    override suspend fun getCharacterList(): KResponse<List<Character>> {
         val resp = getCharactersRemote()
-        val characterList = resp.data ?: loadCached()
+        val data = (resp as? KResponse.Success)?.data
+        val characterList = data ?: loadCached()
         return if (characterList.isEmpty()) {
             resp
```

Comment:
Didn't look at this much but why is the UI handling this? 

A great case will be the data layer handling all API responses (converting them to be presentable for the UI) and sends a response that can be directly consumed by the UI. What do you think?

Reply:
@chizoba this is the data layer though, or I don't really understand what you mean

Reply:
You are currently returning the response from the server (KResponse). Is there no UI model to map to and return? (this can also be done in the domain layer i believe)